raft.tla

```diff
@@ -354,7 +354,8 @@ HandleAppendEntriesRequest(i, j, m) ==
              /\ LET index == m.mprevLogIndex + 1
                 IN \/ \* already done with request
                        /\ \/ m.mentries = << >>
-                          \/ /\ Len(log[i]) >= index
+                          \/ /\ m.mentries /= << >>
```

Comment:
No, it will not.  `\/` constructs a collection of possible actions in one step. for example, `a \/ b \/ c` means `a` or `b` or `c` will be `executed` if it's evaluated to `true`, then produce a new next state. If `a`, `b` and `c` are all `true`, then they will all be `executed` and produce three different next state. 

It's not equal to 
```c
if (eval(a)) { a; }
else if (eval(b)) { b; }
else if (eval(c)) { c; }
```

Reply:
@wgtdkp  ok, thanks for your explanation, but have any official materials to explain it?

Reply:
@Leviathan1995 
Take a look at the [TLA+ video Course](http://lamport.azurewebsites.net/video/videos.html), especially the _**State Machines in TLA+**_ lecture. For advanced learning, you can ref to [_Specifying systems_](https://lamport.azurewebsites.net/tla/book.html) and [_The TLA+ Hyperbook_](https://lamport.azurewebsites.net/tla/hyperbook.html).

More materials can be found at [The TLA Home Page](https://lamport.azurewebsites.net/tla/tla.html#learning).

Reply:
@wgtdkp I find two passages in [Specifying systems](https://lamport.azurewebsites.net/tla/book.html):
![](https://ws1.sinaimg.cn/large/00602bPTgy1ftczg6jfdgj30rb075jui.jpg)
![](https://ws1.sinaimg.cn/large/00602bPTgy1ftczg869kwj30r102tgmy.jpg)

The first passage means that `\/ ` will split computation into `n` subformulas, it accords with your statements, but why the order of the conjuncts will affect that whether or not TLC can evaluate the next-state action.